python/pyspark/worker.py

```diff
@@ -505,6 +505,9 @@ def main(infile, outfile):
             BarrierTaskContext._initialize(boundPort, secret)
         else:
             taskContext = TaskContext._getOrCreate()
+        # Set the task context instance here, so we can get it by TaskContext.get for
+        # both TaskContext and BarrierTaskContext
+        TaskContext._setTaskContext(taskContext)
```

Comment:
We only need to do it in the `if isBarrier` branch, right?

Reply:
updated